### ISSUE TITLE
waiver data is not being passed to Inspec::Reporter::Base#run_data

### ISSUE THREAD

**Issue by zenspider** [inspec/inspec#4467]
Manually running one of our mocks:

```
10000 % ./misc/run.sh exec --reporter json -- test/unit/mock/profiles/waivers/basic | jq .
```

You can see the run_data has nil entries for `:waiver_data`:

```ruby
{:controls=>
  [{:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>0.000837,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"failed",
    :code_desc=>"true should eq false",
    :run_time=>0.008041,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq false",
    :waiver_data=>nil,
    :message=>
     "\n" +
     "expected: false\n" +
     "     got: true\n" +
     "\n" +
     "(compared using ==)\n" +
     "\n" +
     "Diff:\n" +
     "@@ -1,2 +1,2 @@\n" +
     "-false\n" +
     "+true\n"},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>0.000127,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"failed",
    :code_desc=>"true should eq false",
    :run_time=>0.000162,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq false",
    :waiver_data=>nil,
    :message=>
     "\n" +
     "expected: false\n" +
     "     got: true\n" +
     "\n" +
     "(compared using ==)\n" +
     "\n" +
     "Diff:\n" +
     "@@ -1,2 +1,2 @@\n" +
     "-false\n" +
     "+true\n"},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>5.4e-05,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>5.0e-05,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"failed",
    :code_desc=>"true should eq false",
    :run_time=>0.000147,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq false",
    :waiver_data=>nil,
    :message=>
     "\n" +
     "expected: false\n" +
     "     got: true\n" +
     "\n" +
     "(compared using ==)\n" +
     "\n" +
     "Diff:\n" +
     "@@ -1,2 +1,2 @@\n" +
     "-false\n" +
     "+true\n"},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>5.1e-05,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>5.8e-05,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil},
   {:status=>"failed",
    :code_desc=>"true should eq false",
    :run_time=>0.00013,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq false",
    :waiver_data=>nil,
    :message=>
     "\n" +
     "expected: false\n" +
     "     got: true\n" +
     "\n" +
     "(compared using ==)\n" +
     "\n" +
     "Diff:\n" +
     "@@ -1,2 +1,2 @@\n" +
     "-false\n" +
     "+true\n"},
   {:status=>"passed",
    :code_desc=>"true should eq true",
    :run_time=>4.9e-05,
    :start_time=>"2019-09-10T17:23:05-07:00",
    :resource_title=>true,
    :expectation_message=>"should eq true",
    :waiver_data=>nil}],
 :other_checks=>[],
 :profiles=>
  [{:name=>"basic",
    :license=>"Apache-2.0",
    :summary=>"A profile that demonstrates basic usage of the waiver system",
    :version=>"0.1.0",
    :supports=>[{:platform=>"os"}],
    :controls=>
     [{:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"01_not_waivered_passes\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>1},
       :id=>"01_not_waivered_passes",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>0.000837,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"02_not_waivered_fails\" do\n" +
        "  describe(true) { it { should eq false } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>5},
       :id=>"02_not_waivered_fails",
       :results=>
        [{:status=>"failed",
          :code_desc=>"true should eq false",
          :run_time=>0.008041,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq false",
          :waiver_data=>nil,
          :message=>
           "\n" +
           "expected: false\n" +
           "     got: true\n" +
           "\n" +
           "(compared using ==)\n" +
           "\n" +
           "Diff:\n" +
           "@@ -1,2 +1,2 @@\n" +
           "-false\n" +
           "+true\n"}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"03_waivered_no_expiry_not_skipped_passes\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>9},
       :id=>"03_waivered_no_expiry_not_skipped_passes",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>0.000127,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"04_waivered_no_expiry_not_skipped_fails\" do\n" +
        "  describe(true) { it { should eq false } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>13},
       :id=>"04_waivered_no_expiry_not_skipped_fails",
       :results=>
        [{:status=>"failed",
          :code_desc=>"true should eq false",
          :run_time=>0.000162,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq false",
          :waiver_data=>nil,
          :message=>
           "\n" +
           "expected: false\n" +
           "     got: true\n" +
           "\n" +
           "(compared using ==)\n" +
           "\n" +
           "Diff:\n" +
           "@@ -1,2 +1,2 @@\n" +
           "-false\n" +
           "+true\n"}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"05_waivered_no_expiry_skipped\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>17},
       :id=>"05_waivered_no_expiry_skipped",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>5.4e-05,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"06_waivered_expiry_in_past_not_skipped_passes\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>21},
       :id=>"06_waivered_expiry_in_past_not_skipped_passes",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>5.0e-05,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"07_waivered_expiry_in_past_not_skipped_fails\" do\n" +
        "  describe(true) { it { should eq false } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>25},
       :id=>"07_waivered_expiry_in_past_not_skipped_fails",
       :results=>
        [{:status=>"failed",
          :code_desc=>"true should eq false",
          :run_time=>0.000147,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq false",
          :waiver_data=>nil,
          :message=>
           "\n" +
           "expected: false\n" +
           "     got: true\n" +
           "\n" +
           "(compared using ==)\n" +
           "\n" +
           "Diff:\n" +
           "@@ -1,2 +1,2 @@\n" +
           "-false\n" +
           "+true\n"}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"08_waivered_expiry_in_past_skipped\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>29},
       :id=>"08_waivered_expiry_in_past_skipped",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>5.1e-05,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"09_waivered_expiry_in_future_not_skipped_passes\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>33},
       :id=>"09_waivered_expiry_in_future_not_skipped_passes",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>5.8e-05,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"10_waivered_expiry_in_future_not_skipped_fails\" do\n" +
        "  describe(true) { it { should eq false } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>37},
       :id=>"10_waivered_expiry_in_future_not_skipped_fails",
       :results=>
        [{:status=>"failed",
          :code_desc=>"true should eq false",
          :run_time=>0.00013,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq false",
          :waiver_data=>nil,
          :message=>
           "\n" +
           "expected: false\n" +
           "     got: true\n" +
           "\n" +
           "(compared using ==)\n" +
           "\n" +
           "Diff:\n" +
           "@@ -1,2 +1,2 @@\n" +
           "-false\n" +
           "+true\n"}]},
      {:title=>nil,
       :desc=>nil,
       :descriptions=>{},
       :impact=>0.5,
       :refs=>[],
       :tags=>{},
       :code=>
        "control \"11_waivered_expiry_in_future_skipped\" do\n" +
        "  describe(true) { it { should eq true } }\n" +
        "end\n",
       :source_location=>
        {:ref=>"test/unit/mock/profiles/waivers/basic/controls/basic.rb",
         :line=>41},
       :id=>"11_waivered_expiry_in_future_skipped",
       :results=>
        [{:status=>"passed",
          :code_desc=>"true should eq true",
          :run_time=>4.9e-05,
          :start_time=>"2019-09-10T17:23:05-07:00",
          :resource_title=>true,
          :expectation_message=>"should eq true",
          :waiver_data=>nil}]}],
    :groups=>
     [{:title=>nil,
       :controls=>
        ["01_not_waivered_passes",
         "02_not_waivered_fails",
         "03_waivered_no_expiry_not_skipped_passes",
         "04_waivered_no_expiry_not_skipped_fails",
         "05_waivered_no_expiry_skipped",
         "06_waivered_expiry_in_past_not_skipped_passes",
         "07_waivered_expiry_in_past_not_skipped_fails",
         "08_waivered_expiry_in_past_skipped",
         "09_waivered_expiry_in_future_not_skipped_passes",
         "10_waivered_expiry_in_future_not_skipped_fails",
         "11_waivered_expiry_in_future_skipped"],
       :id=>"controls/basic.rb"}],
    :inputs=>[],
    :sha256=>
     "b370faa3a481aa4e84d6214bd6b5968c0c33460e0d03c43bd289b11764b86ad8",
    :status=>"loaded"}],
 :platform=>{:name=>"mac_os_x", :release=>"18.7.0", :target=>"local://"},
 :version=>"4.16.10",
 :statistics=>
  {:duration=>0.012921,
   :controls=>
    {:total=>11,
     :passed=>{:total=>7},
     :skipped=>{:total=>0},
     :failed=>{:total=>4}}}}
```

And that shows up in the json report only once as nil on controls:

```json
{
  "platform": {
    "name": "mac_os_x",
    "release": "18.7.0"
  },
  "profiles": [
    {
      "name": "basic",
      "version": "0.1.0",
      "sha256": "b370faa3a481aa4e84d6214bd6b5968c0c33460e0d03c43bd289b11764b86ad8",
      "title": null,
      "maintainer": null,
      "summary": "A profile that demonstrates basic usage of the waiver system",
      "license": "Apache-2.0",
      "copyright": null,
      "copyright_email": null,
      "supports": [
        {
          "platform": "os"
        }
      ],
      "attributes": [],
      "parent_profile": null,
      "depends": null,
      "groups": [
        {
          "id": "controls/basic.rb",
          "controls": [
            "01_not_waivered_passes",
            "02_not_waivered_fails",
            "03_waivered_no_expiry_not_skipped_passes",
            "04_waivered_no_expiry_not_skipped_fails",
            "05_waivered_no_expiry_skipped",
            "06_waivered_expiry_in_past_not_skipped_passes",
            "07_waivered_expiry_in_past_not_skipped_fails",
            "08_waivered_expiry_in_past_skipped",
            "09_waivered_expiry_in_future_not_skipped_passes",
            "10_waivered_expiry_in_future_not_skipped_fails",
            "11_waivered_expiry_in_future_skipped"
          ]
        }
      ],
      "controls": [
        {
          "id": "01_not_waivered_passes",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"01_not_waivered_passes\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 1,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 0.000837,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "02_not_waivered_fails",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"02_not_waivered_fails\" do\n  describe(true) { it { should eq false } }\nend\n",
          "source_location": {
            "line": 5,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "failed",
              "code_desc": "true should eq false",
              "run_time": 0.008041,
              "start_time": "2019-09-10T17:23:05-07:00",
              "message": "\nexpected: false\n     got: true\n\n(compared using ==)\n\nDiff:\n@@ -1,2 +1,2 @@\n-false\n+true\n"
            }
          ]
        },
        {
          "id": "03_waivered_no_expiry_not_skipped_passes",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"03_waivered_no_expiry_not_skipped_passes\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 9,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 0.000127,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "04_waivered_no_expiry_not_skipped_fails",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"04_waivered_no_expiry_not_skipped_fails\" do\n  describe(true) { it { should eq false } }\nend\n",
          "source_location": {
            "line": 13,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "failed",
              "code_desc": "true should eq false",
              "run_time": 0.000162,
              "start_time": "2019-09-10T17:23:05-07:00",
              "message": "\nexpected: false\n     got: true\n\n(compared using ==)\n\nDiff:\n@@ -1,2 +1,2 @@\n-false\n+true\n"
            }
          ]
        },
        {
          "id": "05_waivered_no_expiry_skipped",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"05_waivered_no_expiry_skipped\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 17,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 5.4e-05,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "06_waivered_expiry_in_past_not_skipped_passes",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"06_waivered_expiry_in_past_not_skipped_passes\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 21,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 5e-05,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "07_waivered_expiry_in_past_not_skipped_fails",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"07_waivered_expiry_in_past_not_skipped_fails\" do\n  describe(true) { it { should eq false } }\nend\n",
          "source_location": {
            "line": 25,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "failed",
              "code_desc": "true should eq false",
              "run_time": 0.000147,
              "start_time": "2019-09-10T17:23:05-07:00",
              "message": "\nexpected: false\n     got: true\n\n(compared using ==)\n\nDiff:\n@@ -1,2 +1,2 @@\n-false\n+true\n"
            }
          ]
        },
        {
          "id": "08_waivered_expiry_in_past_skipped",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"08_waivered_expiry_in_past_skipped\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 29,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 5.1e-05,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "09_waivered_expiry_in_future_not_skipped_passes",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"09_waivered_expiry_in_future_not_skipped_passes\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 33,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 5.8e-05,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        },
        {
          "id": "10_waivered_expiry_in_future_not_skipped_fails",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"10_waivered_expiry_in_future_not_skipped_fails\" do\n  describe(true) { it { should eq false } }\nend\n",
          "source_location": {
            "line": 37,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "failed",
              "code_desc": "true should eq false",
              "run_time": 0.00013,
              "start_time": "2019-09-10T17:23:05-07:00",
              "message": "\nexpected: false\n     got: true\n\n(compared using ==)\n\nDiff:\n@@ -1,2 +1,2 @@\n-false\n+true\n"
            }
          ]
        },
        {
          "id": "11_waivered_expiry_in_future_skipped",
          "title": null,
          "desc": null,
          "descriptions": [],
          "impact": 0.5,
          "refs": [],
          "tags": {},
          "code": "control \"11_waivered_expiry_in_future_skipped\" do\n  describe(true) { it { should eq true } }\nend\n",
          "source_location": {
            "line": 41,
            "ref": "test/unit/mock/profiles/waivers/basic/controls/basic.rb"
          },
          "results": [
            {
              "status": "passed",
              "code_desc": "true should eq true",
              "run_time": 4.9e-05,
              "start_time": "2019-09-10T17:23:05-07:00"
            }
          ]
        }
      ],
      "status": "loaded",
      "skip_message": null,
      "waiver_data": null
    }
  ],
  "statistics": {
    "duration": 0.012921
  },
  "version": "4.16.10"
}
```

So I think some glue is missing on phase 2 of waivers, blocking #2445

**Comment by clintoncwolfe** [comment]
I think it's behaving slightly out of spec, but workably so.

Based on the given invocation - which omits waiver data entirely:

```
bundle exec inspec exec --reporter json -- test/unit/mock/profiles/waivers/basic | jq .
```

there *should* be no waiver_data entries. Instead, the entries are improperly present but empty.

Adding a waiver file changes behavior (I'm changing arg order to be more typical):

```
bundle exec inspec exec test/unit/mock/profiles/waivers/basic --reporter json --waiver-file test/unit/mock/profiles/waivers/basic/files/waiver.yaml | jq .
```

In a debugger (I injected at https://github.com/inspec/inspec/blob/e0e61155d6d9b2dd146410994bdef3b0048fac5c/lib/inspec/reporters/json.rb#L42), r[:waiver_data] is always present, nil if the control was not waivered (which is out of spec), and populated if the control had waiver information associated with it. In the "basic" profile used in testing, control index 0 and 1 are not waivered, but 2 is:

```
(byebug) r[:waiver_data]
{"justification"=>"Unassailable thinking", "skip"=>false, "skipped_due_to_waiver"=>false, "message"=>""}
```

If needed, I can make a fix so that `r[:waiver_data]` only appears if it is populated.

**Comment by clintoncwolfe** [comment]
I believe this turned out to be a temporary issue caused by another issue - once that was resolved, and with proper invocation (above) the data was found, and waivers 3 was merged.
